config.go

```diff
@@ -153,11 +164,30 @@ func (c *ClientConfig) setDefaults(cfg *Config) {
 	c.Timeout = cfg.ClientTimeout
 }
 
-func (c *ClientConfig) validate() error {
+func (c *ClientConfig) validate(minLifetime time.Duration, logger *logrus.Entry) error {
 	if c.Key == "" {
 		return errors.New("No key in config")
 	}
 
+	keyPair, err := tls.LoadX509KeyPair(c.Cert, c.Key)
+	if err != nil {
+		return fmt.Errorf("Invalid key/cert in config: %v", err)
+	}
+
+	leaf, err := x509.ParseCertificate(keyPair.Certificate[0])
+	if err != nil {
+		return fmt.Errorf("Invalid key/cert in config: %v", err)
+	}
+
+	// Want to make sure cert has a minimum lifetime as given by config.
+	// If expiry date minus minimum lifetime is after the current point
+	// in time, we're ok, otherwise not.
+	if leaf.NotAfter.Add(-minLifetime).Before(time.Now()) {
+		logger.Warnf(
+			"Certificate for client '%s', in file(s) '%s'/'%s' is expiring in less than %s",
```

Comment:
Nit: Is it intentional to print the certificate filename rather than a client name in the `client '%s'` section?

Reply:
Nope, that's a bug!